packages/prop-controllers/src/link/link.test.ts

```diff
@@ -0,0 +1,295 @@
+import { describe, expect, test } from '@jest/globals'
+
+import { ControlDataTypeKey } from '../prop-controllers'
+import {
+  LinkPropControllerData,
+  LinkPropControllerDataV0,
+  LinkPropControllerDataV1,
+  LinkPropControllerDataV1Type,
+  copyLinkPropControllerData,
+  getLinkPropControllerPageIds,
+  getLinkPropControllerValue,
+} from './link'
+import { createReplacementContext } from '../utils/utils'
+
+describe('LinkPropController', () => {
+  describe('getLinkPropControllerValue', () => {
+    test('returns value for LinkPropControllerDataV1Type', () => {
+      // Arrange
+      const data: LinkPropControllerDataV1 = {
+        [ControlDataTypeKey]: LinkPropControllerDataV1Type,
+        value: {
+          type: 'OPEN_PAGE',
+          payload: {
+            pageId: 'UGFnZTpmNTdmMjQ2MS0wMGY3LTQzZWUtYmIwOS03ODdiNTUyYzUyYWQ=',
+            openInNewTab: false,
+          },
+        },
+      }
+
+      // Act
+      const result = getLinkPropControllerValue(data)
+
+      // Assert
+      expect(result).toEqual(data.value)
+    })
+
+    test('returns value for LinkPropControllerDataV0 data', () => {
+      // Arrange
+      const data: LinkPropControllerDataV0 = {
+        type: 'OPEN_PAGE',
+        payload: {
+          pageId: 'UGFnZTpmNTdmMjQ2MS0wMGY3LTQzZWUtYmIwOS03ODdiNTUyYzUyYWQ=',
+          openInNewTab: false,
+        },
+      }
+
+      // Act
+      const result = getLinkPropControllerValue(data)
+
+      // Assert
+      expect(result).toEqual(data)
+    })
+
+    test('returns value for unknown data', () => {
+      // Arrange
+      const data = {
+        test: 'unknown',
+      }
+
+      // Act
+      // @ts-expect-error: invalid data
+      const result = getLinkPropControllerValue(data)
+
+      // Assert
+      expect(result).toEqual(data)
+    })
```

Comment:
This is an interesting test case. One could argue that this should be a parsing error.

Reply:
That's true. I think the problem is that right now we don't have a way to gracefully handle parse errors in the `PropsValue`

Reply:
Yeah, maybe as we evolve this we can do a better job of handling bad data in these functions. I'll see what I can do when working with the Shadows prop controller